packages/flutter/lib/src/widgets/gesture_detector.dart

```diff
@@ -54,20 +54,51 @@ class GestureDetector extends StatefulComponent {
 
   final Widget child;
 
+  /// A pointer that might cause a tap has contacted the screen at a particular
+  /// location.
   final GestureTapDownCallback onTapDown;
+
+  /// A pointer that will trigger a tap has stopped contacting the screen at a
+  /// particular location.
   final GestureTapDownCallback onTapUp;
+
+  /// A tap has occurred.
   final GestureTapCallback onTap;
+
+  /// The pointer that previously triggered the [onTapDown] will not end up
+  /// causing a tap.
   final GestureTapCancelCallback onTapCancel;
+
+  /// The user has tapped the screen at the same location twice in quick
+  /// succession.
   final GestureTapCallback onDoubleTap;
 
+  /// A pointer has remained in contact with the screen at the same location for
+  /// a long period of time.
   final GestureLongPressCallback onLongPress;
 
+  /// A pointer has contacted the screen and might begin to move vertically.
   final GestureDragStartCallback onVerticalDragStart;
+
+  /// A pointer that is in contact with the screen and moving vertically has
+  /// moved again in the vertical direction.
```

Comment:
I'm a bit confused by the "again" here, since the ...Start callback doesn't indicate motion. What event is sent on the very first vertical motion? A slight rephrase might be in order here, though I'm not sure exactly what to suggest.